doc/auth.md

```diff
@@ -0,0 +1,320 @@
+# Kuadrant Auth
+
+A Kuadrant AuthPolicy custom resource:
+
+1. Allows it to target Gateway API networking resources such as [HTTPRoutes](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRoute) and [Gateways](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.Gateway), using these resources to obtain additional context, i.e., which traffic workload (HTTP attributes, hostnames, user attributes, etc) to enforce auth.
+2. Allows to specify which specific subsets of the targeted network resource to apply the auth rules to.
+3. Abstracts the details of the underlying external authorization protocol and configuration resources, that have a much broader remit and surface area.
+4. Supports cluster operators to set overrides (soon) and defaults that govern what can be done at the lower levels.
+
+## How it works
+
+### Envoy's External Authorization Protocol
+
+Kuadrant's Auth implementation relies on the Envoy's [External Authorization](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/ext_authz_filter) protocol. The workflow per request goes:
+1. On incoming request, the gateway checks the matching rules for enforcing the auth rules, as stated in the AuthPolicy custom resources and targeted Gateway API networking objects
+2. If the request matches, the gateway sends one [CheckRequest](https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/auth/v3/external_auth.proto#envoy-v3-api-msg-service-auth-v3-checkrequest) to the external auth service ("Authorino").
+3. The external auth service responds with a [CheckResponse](https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/auth/v3/external_auth.proto#service-auth-v3-checkresponse) back to the gateway with either an `OK` or `DENIED` response code.
+
+An AuthPolicy and its targeted Gateway API networking resource contain all the statements to configure both the ingress gateway and the external auth service.
+
+### The AuthPolicy custom resource
+
+#### Overview
+
+The `AuthPolicy` spec includes the following parts:
+
+* A reference to an existing Gateway API resource (`spec.targetRef`)
+* Authentication/authorization scheme (`spec.rules`)
+* Top-level route selectors (`spec.routeSelectors`)
+* Top-level additional conditions (`spec.when`)
+* List of stored named patterns (`spec.patterns`)
+
+The auth scheme specify rules for:
+* Authentication (`spec.rules.authentication`)
+* External auth metadata fetching (`spec.rules.metadata`)
+* Authorization (`spec.rules.authorization`)
+* Custom response items (`spec.rules.response`)
+* Callbacks (`spec.rules.callbacks`)
+
+where each auth rule can declare specific `routeSelectors` and `when` conditions for the rule to apply.
```

Comment:
Very surprised I spotted this.
```suggestion
Where each auth rule can declare specific `routeSelectors` and `when` conditions for the rule to apply.
```

Reply:
The lower case was intentional.

I know this is not scientific writing, but it's a common pattern in texts where you have formalisation of concepts with blocks in between sentences.

The sentence that starts with "where" should be read in continuity to "The auth scheme specify rules for..."; otherwise it loses context.

But perhaps the following can please us both 🙂:

```suggestion
Each auth rule can declare specific `routeSelectors` and `when` conditions for the rule to apply.
```